### ISSUE TITLE
zIndex doesn't work when the map moves in iOS 11

### ISSUE THREAD

**Issue by KarlosQ** [react-community/react-native-maps#1671]
The yellow marker is ever in foreground in others version or in Android.

In iOS 11 only works when you press in the marker but doesn't work when change the zIndex by state for example.

Somebody know how I can fix this?

![screen shot 2017-11-29 at 13 24 25](https://user-images.githubusercontent.com/15233913/33375137-3f814262-d509-11e7-8286-89414c920b79.png)

My MapView code: 

````
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Dimensions
} from 'react-native';
const { width, height } = Dimensions.get('window');
import MapView, { PROVIDER_GOOGLE, PROVIDER_DEFAULT } from 'react-native-maps';
const { Marker } = MapView;
export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0
    };
    this.markers = [
      { id: 1, latitude: 43.152, longitude: -1.30051 },
      { id: 2, latitude: 43.1525, longitude: -1.30201 },
      { id: 3, latitude: 43.15264, longitude: -1.30025 },
      { id: 5, latitude: 43.1527, longitude: -1.30201 },
      { id: 6, latitude: 43.1525, longitude: -1.3001 },
      { id: 7, latitude: 43.1527, longitude: -1.30043 },
      { id: 8, latitude: 43.1526, longitude: -1.300531 },
      { id: 9, latitude: 43.1525, longitude: -1.30011 },
      { id: 10, latitude: 43.1521, longitude: -1.30062 },
      { id: 4, latitude: 43.152, longitude: -1.30083 }
    ];
  }
  render() {
    return (
      <View style={styles.container}>
        <MapView
          provider={PROVIDER_DEFAULT}
          style={{ flex: 1 }}
          initialRegion={{
            latitude: 43.153,
            longitude: -1.3,
            longitudeDelta: 20,
            latitudeDelta: 15
          }}
          minZoomLevel={15}
          ref={ref => { this.map = ref }}>
          {this.markers.map((marker, i) => {
            const selected = this.state.index == i;
            return (
              <Marker
                key={'m' + marker.id}
                coordinate={{ latitude: marker.latitude, longitude: marker.longitude }}
                onPress={() => this.setState({ index: i })}
                zIndex={selected ? 2 : 1}>
                <View style={styles.marker}>
                  <View style={styles.bubbleContainer}>
                    <View style={[styles.bubble, selected && styles.selected]}>
                      <Text style={styles.txt}>test</Text>
                    </View>
                  </View>
                </View>
              </Marker>
            );
          })}
        </MapView>
        <TouchableOpacity onPress={() => this.setState({ index: Math.round(Math.random() * 10) })}>
          <Text>Random select</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    height
  },
  marker: {
    width: 50
  },
  bubbleContainer: {
    flexDirection: 'column',
    alignSelf: 'flex-start',
    backgroundColor: 'transparent',
    height: 45,
  },
  bubble: {
    flex: 0,
    flexDirection: 'row',
    alignSelf: 'flex-start',
    justifyContent: 'center',
    padding: 10,
    borderRadius: 10,
    borderColor: '#D1D1D1',
    borderWidth: 1,
    width: 49,
    backgroundColor: '#fff',
  },
  txt: {
    color: '#000',
  },
  selected: {
    backgroundColor: '#F7FF00'
  },
});
````

RN: 0.48.4 / 0.50.4
RN-maps: 0.16.4 / 0.18.1
OS: iOS 11 / 11.1

**Comment by germanurrus** [comment]
I have the same issue, but after you move the map, it seems that the zIndex starts applying...I'm still investigating for a solution...

**Comment by KimAM** [comment]
I have this problem in iOS 11.1 Beta as well. zIndex never applies, even after I move the map

**Comment by danieljvdm** [comment]
Any updates? It's not working with this code:

```
<MapView.Marker
  identifier={listing.uuid}
  key={listing.uuid}
  coordinate={listing.coordinate}
  ref={this._markerRef}
  onPress={this._onMarkerPress}
  style={isSelected ? { zIndex: 1 } : null}>
  <MarkerView rent="$4,500" isSelected={isSelected} />
</MapView.Marker>
```

**Comment by danieljvdm** [comment]
I tinkered around a bit with some of the native code but didn't achieve anything. One thing I noticed though is zIndex _does_ seem to work it just doesn't work when markers are too close together.

**Comment by testxin** [comment]
iphone 6 plus ios 11 the same quesion.  has solution?

**Comment by KarlosQ** [comment]
@testxin not yet... 

@lelandrichardson @christopherdro Can you help us, please? :3

**Comment by HerbertLim** [comment]
I have same issue on my iOS. Found that zIndex works well at Android.

**Comment by KarlosQ** [comment]
Somebody try this solution? [](https://stackoverflow.com/questions/46518725/mapkit-mkmapview-zposition-does-not-work-anymore-on-ios11)

**Comment by jnrepo** [comment]
Seeing this issue on the latest 
react-native `0.50.3`
react-native maps `0.17.3`
iOS: 11.1

- zIndex is not honored and the marker is not at the top. 
- Clicking on a not selected marker will actually render behind other markers as well now

**Comment by KarlosQ** [comment]
any news?

**Comment by reyalpsirc** [comment]
This seems to be related to the same thing as [https://stackoverflow.com/questions/46518725/mapkit-mkmapview-zposition-does-not-work-anymore-on-ios11](https://stackoverflow.com/questions/46518725/mapkit-mkmapview-zposition-does-not-work-anymore-on-ios11)

**Comment by mafiusu** [comment]
Any solution how I can achieve that my custom position marker is always behind the rendered Markers for the Map?

**Comment by TarasDumych** [comment]
@mafiusu => @KarlosQ solution works for me. But you also should set custom marker via `image` prop.

**Comment by mmiani** [comment]
Still having this issue on 
react-native 0.51
react-native maps 0.21
OS: iOS 11 / 11.3

**Comment by dahiri-farid** [comment]
I have made a fork with a fix.
https://github.com/GET-A-WAY/react-native-maps

**Comment by alvelig** [comment]
closed by #2359
